src/topp/PhosphoScoring.cpp

```diff
@@ -93,15 +92,150 @@ class TOPPPhosphoScoring :
 
 protected:
 
+  // spectrum must not contain 0 intensity peaks and must be sorted by m/z
+  template <typename SpectrumType>
+  static void deisotopeAndSingleChargeMSSpectrum_(SpectrumType& in, Int min_charge, Int max_charge, double fragment_tolerance, bool fragment_unit_ppm, bool keep_only_deisotoped = false, Size min_isopeaks = 3, Size max_isopeaks = 10, bool make_single_charged = true)
```

Comment:
this might be related to #974 and this http://sourceforge.net/p/open-ms/mailman/message/30818069/ on the mailing list ,so it is a long-standing issue and might also merit its own issue